src/components/canvas/canvas.js

```diff
@@ -35,27 +35,39 @@ export default class Canvas extends React.Component {
     metadata: {},
   };
 
-  componentDidMount() {
-    const { onUpdate } = this.props;
-    const bus = new CanvasEventBus();
+  onDragStart = (metadata) => bus.emit('dragstart', metadata);
+  onDragEnd = (metadata) => {
+    const { source, destination } = metadata;
+    destination && source.droppableId === destination.droppableId
+      ? bus.emit('reorder', metadata)
+      : bus.emit('move', metadata);
+    bus.emit('dragend', {});
+  };
 
+  componentDidMount() {
     this.eventsSubscription = bus.getEventsStream(ReactDOM.findDOMNode(this)).subscribe(([event, metadata]) => {
       switch (event) {
-        case 'drop':
-          return onUpdate(createNode(this.props.layout, metadata));
+        case 'dragstart':
```

Comment:
case 'dragstart',
case 'dragover',
case 'dragend':
    return this.setState({ metadata });

Reply:
#prettier